File: output_at_5/claude-opus/sample_2/c/c/Struct/spouse.c
#include "stdlib.h"

struct person {
  char* name;
  struct person* spouse;
};

/*@
predicate person(struct person* p;) = p->name |-> _ &*& p->spouse |-> ?s &*& (s == 0 || true);
@*/

struct person *create_person()
  //@ requires true;
  //@ ensures person(result);
{
  struct person *p = malloc(sizeof(struct person));
  if(p == 0) abort();
  p->spouse = 0;
  //@ close person(p);
  return p;
}

void marry(struct person *this, struct person *other)
  //@ requires person(this) &*& person(other);
  //@ ensures person(this) &*& person(other);
{
  //@ open person(this);
  //@ open person(other);
  this->spouse = other;
  other->spouse = this;
  //@ close person(this);
  //@ close person(other);
}

struct person* person_get_spouse(struct person* this)
  //@ requires person(this);
  //@ ensures person(this) &*& (result == 0 ? true : person(result));
{
  //@ open person(this);
  struct person* s = this->spouse;
  //@ close person(this);
  return s;
}

void divorce(struct person* this)
  //@ requires person(this);
  //@ ensures person(this);
{
  //@ open person(this);
  if(this->spouse != 0) {
    //@ open person(this->spouse);
    this->spouse->spouse = 0;
    //@ close person(this->spouse);
  }
  this->spouse = 0;
  //@ close person(this);
}

void die(struct person *this)
  //@ requires person(this);
  //@ ensures true;
{
  //@ open person(this);
  if(this->spouse != 0) {
    //@ open person(this->spouse);
    this->spouse->spouse = 0;
    //@ close person(this->spouse);
  }
  free(this); 
}

int main() 
  //@ requires true;
  //@ ensures true;
{
  struct person* alice = create_person();
  struct person* bob = create_person();
  struct person* eve = 0;
  marry(alice, bob);
  eve = create_person();
  divorce(bob);
  marry(bob, eve);
  die(alice);
  die(bob);
  die(eve);
  return 0;
}

File: output_at_5/gpt52/sample_1/java/java/Linklist/Spouse.java
class Person {

    protected Person spouse;

    /*@
    predicate person(Person p; Person s) = p.spouse |-> s;
    @*/
    
    public  void spouse_symm()
        
        
    //@ requires person(this, ?s);
    //@ ensures person(this, s);
    {
        
        
        
    }

    public Person()
        
        
    //@ requires true;
    //@ ensures person(this, null);
    {
        //@ close person(this, null);
    }
    
    public Person getSpouse()
        
        
    //@ requires person(this, ?s);
    //@ ensures person(this, s) &*& result == s;
    {
        //@ open person(this, s);
        Person result_ = spouse;
        //@ close person(this, s);
        return result_;
        
    }
    
    protected void setSpouse(Person other)
        
        
    //@ requires person(this, _) &*& person(other, _);
    //@ ensures person(this, other) &*& person(other, this);
    {
        //@ open person(this, _);
        //@ open person(other, _);
        spouse = other;
        other.spouse = this;
        //@ close person(this, other);
        //@ close person(other, this);
        
    }
    
    protected void clearSpouse()
        
        
    //@ requires person(this, ?s) &*& s != null &*& person(s, this);
    //@ ensures person(this, null) &*& person(s, null);
    {
        //@ open person(this, s);
        //@ open person(s, this);
        spouse.spouse = null;
        spouse = null;
        //@ close person(s, null);
        //@ close person(this, null);
        
    }
    
    void marry(Person other)
        
        
    //@ requires person(this, _) &*& person(other, _);
    //@ ensures person(this, other) &*& person(other, this);
    {
        other.setSpouse(this);
        
    }
    
    void divorce()
        
        
    //@ requires person(this, ?s) &*& s != null &*& person(s, this);
    //@ ensures person(this, null) &*& person(s, null);
    {
        //@ open person(this, s);
        spouse.clearSpouse();
        //@ close person(this, null);
        
    }

}

class Program {

    public static void main(String[] args)
        
        
    //@ requires true;
    //@ ensures true;
    {
        Person a = new Person();
        Person b = new Person();
        //@ open Person.person(a, null);
        //@ open Person.person(b, null);
        a.marry(b);
        //@ close Person.person(a, b);
        //@ close Person.person(b, a);
        //@ open Person.person(b, a);
        //@ open Person.person(a, b);
        b.divorce();
        //@ close Person.person(b, null);
        //@ close Person.person(a, null);
    }

}